pkg/cache/partition_info.go

```diff
@@ -169,6 +169,18 @@ func (pi *PartitionInfo) addNewNode(node *NodeInfo, existingAllocations []*si.Al
             zap.String("nodeId", node.NodeId),
             zap.Int("existingAllocations", len(existingAllocations)))
         for _, alloc := range existingAllocations {
+            // if this is to start processing existing allocations for an app,
+            // transit app's state from Accepted to Running
+            if app, ok := pi.applications[alloc.ApplicationId]; ok {
```

Comment:
Shouldn't this happen after we have successfully added the allocation(s)?
We check the outcome of the allocation and reject the node if the allocation fails. I think we should only move the apps to running when the node addition has succeeded just before we update the metric that the node is active.

Reply:
We can do that. I wasn't having a strong preference about the time point, what you said makes sense, I just revised the logic a bit to do that. Thx